### ISSUE TITLE
fatal: your current branch 'master' does not have any commits yet 

### ISSUE THREAD

**Issue by febeks** [git-ftp/git-ftp#560]
I am using bitbucket, where I have `master` and `dev` branches, currently `dev` is set as default branch.
I have only initial commit in `master` branch (readme.md).

Directory structure of my project is:
dev/backoffice (angular app)
dev/backend (laravel app)

I am using Jenkins to build my app (for now only angular app) and then deploy to ftp using these commands:

```
cd backoffice

git config git-ftp.url ftp.website.com
git config git-ftp.user username
git config git-ftp.password password

npm run ng -- build 

git ftp init
```
and when it builds the app and starts deploying, this is what happens:
```
+ git ftp init
fatal: your current branch 'master' does not have any commits yet
There are no files to sync.
Last deployment changed from  to .
Finished: SUCCESS
```

I dont do anything with `master` branch. The entire jenkins build is within `dev`.

**Comment by mkllnk** [comment]
I'm a bit confused. The error message suggests that you need to checkout the `dev` branch first:
```
git checkout dev
npm run ng -- build
git ftp init
```

But if you aren't on the `dev` branch already, then you wouldn't be able to compile your code. So I'm not sure why that step is working for you.

Anyway, Git-ftp is most useful for files that you actually keep under version control with Git. Since you are using Jenkins, you could create a separate Git repository for your compiled files on the Jenkins server and run Git-ftp from in there.

**Comment by febeks** [comment]
I am in the dev branch already because in the Jenkins job settings under source management I set it to pull from Dev branch from my bitbucket repository. I tried to  checkout Dev branch but it says dev doesn't exist or something like that.

What do you mean by creating separate git directory for compiled files?

**Comment by mkllnk** [comment]
It sounds like you keep your source code in a Git repository and then you compile it with `ng`. The compiled files are not version controlled in Git, right?

If you don't have those files under version control, Git doesn't know which files changes since your last deploy and you always need to upload all files. But if you created a Git repository for your compiled files you can use Git-ftp within that repository to upload only changed files.

**Comment by febeks** [comment]
And is that a normal way to do this? Because we push code to bitbucket repo which is not build. Then I run `ng` etc and it creates build files and uploads them to the hosting... I got it to work somehow but now I am getting SHA1 error every time...very frustrating....I tried all options but cant get it to work...

**Comment by mkllnk** [comment]
@febeks Are you using Bitbuckets Pipelines? They don't clone the whole Git repository and don't know all commits. I'm not sure which command is best but maybe a git fetch would help.

**Comment by febeks** [comment]
Yes I'm using pipelines...I have no idea what would be best.. anyone who
has working bitbucket pipelines with git ftp?

**Comment by ecanro** [comment]
Same issue:
`<Microsoft Windows [Version 10.0.18363.900]`
`(c) 2019 Microsoft Corporation. Todos os direitos reservados.`
`D:\Apps\PHP\task-managerV1>git init`
`Initialized empty Git repository in D:/Apps/PHP/task-managerV1/.git/`
use git add:
`D:\Apps\PHP\task-managerV1>git add .`
note not any messages log receive
use git commit:
`D:\Apps\PHP\task-managerV1>git commit -m "firts commit"`
note not any messages log receive
use git log ._.
`D:\Apps\PHP\task-managerV1>git log`
`fatal: your current branch 'master' does not have any commits yet`
`D:\Apps\PHP\task-managerV1>git status`
`On branch master`
`No commits yet`
`Untracked files:`
 ` (use "git add <file>..." to include in what will be committed)`
        `.editorconfig`
         .
         .
         .
       ` .Nfiles`
`nothing added to commit but untracked files present (use "git add" to track)`
in output console
`fatal: your current branch 'master' does not have any commits yet`
git 2.27 in Windows 10 home

**Comment by nsengiyumv33** [comment]
> Yes I'm using pipelines...I have no idea what would be best.. anyone who has working bitbucket pipelines with git ftp?

try to use git fetch, may be it will work
